dask_expr/_shuffle.py

```diff
@@ -740,6 +740,29 @@ def _lower(self):
         return SetPartition(self.frame, self._other, self.drop, divisions)
 
     def _simplify_up(self, parent):
+        from dask_expr._expr import Head, Tail
+
+        # TODO, handle setting index with other frame
+        if (
+            isinstance(parent, Head)
+            and isinstance(self._other, (int, str))
```

Comment:
Could we use the ``is_scalar`` check from dask_expr._util?

Reply:
```
(Pdb) pp self._other
df['x']
(Pdb) pp is_scalar(self._other)
True
```

Something is wrong here.   See also https://github.com/dask-contrib/dask-expr/issues/298

Reply:
I'll take a look, this worries me a bit